File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.py
def solution(players, callings):
    answer = []
    dic = {}
    for idx, val in enumerate(players):
        dic[val] = idx

    for player in callings:
        p1, p2 = dic[player] - 1, dic[player]
        dic[players[p1]] = p2
        dic[players[p2]] = p1
        players[p2],players[p1] = players[p1],players[p2]
    
    return players

File: 프로그래머스/unrated/178871. 달리기 경주/달리기 경주.js
function solution(players, callings) {
    const rank = {}
    
    for(let i=0; i < players.length; i++){
        rank[players[i]] = i
    }
    
    for(call of callings){
        const call_rank = rank[call]
        const change_player = players[call_rank - 1]
        
        players[call_rank] = players[call_rank - 1]
        players[call_rank - 1] = call
        
        rank[change_player] += 1
        rank[call] -=1
    }
    
    return players
}